pyproject.toml

```diff
@@ -3,7 +3,7 @@ requires = [
     "Cython>=0.29.23",
     # Building against NPY2 will support both NPY1 and NPY2
     # https://numpy.org/devdocs/dev/depending_on_numpy.html#build-time-dependency
-    "numpy>=2.1.0",
+    "numpy>=2.0.1",
```

Comment:
`pymatgen` in the past has followed `numpy` in what Python versions are supported. so if `numpy` already dropped Python 3.9 (earlier than I expected), I guess `pymatgen` should as well @shyuep @mkhorton?

Reply:
Yes https://numpy.org/news/ (I thought the [EOL of Python 3.9 would be 2025-10](https://devguide.python.org/versions/)):
>18 Aug, 2024 – NumPy 2.1.0 provides support for Python 3.13 and drops support for Python 3.9.

Reply:
Interesting, it looks like numpy following another protocol other than just Python EOL to determine Python version to support/drop https://github.com/numpy/numpy/issues/24932

Reply:
I will resolve this for a separate PR